api/pkg/apis/v1alpha1/providers/target/piccolo/piccolo_test.go

```diff
@@ -0,0 +1,364 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ * SPDX-License-Identifier: MIT
+ */
+
+package piccolo
+
+import (
+	"context"
+	"os"
+	"testing"
+
+	"github.com/eclipse-symphony/symphony/api/pkg/apis/v1alpha1/model"
+	"github.com/eclipse-symphony/symphony/api/pkg/apis/v1alpha1/providers/target/conformance"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestPiccoloTargetProviderConfigFromMapNil(t *testing.T) {
+	_, err := PiccoloTargetProviderConfigFromMap(nil)
+	assert.Nil(t, err)
+}
+func TestPiccoloTargetProviderConfigFromMapEmpty(t *testing.T) {
+	_, err := PiccoloTargetProviderConfigFromMap(map[string]string{})
+	assert.Nil(t, err)
+}
+func TestPiccoloTargetProviderInitEmptyConfig(t *testing.T) {
+	config := PiccoloTargetProviderConfig{}
+	provider := PiccoloTargetProvider{}
+	err := provider.Init(config)
+	assert.Nil(t, err)
+}
+func TestInitWithMap(t *testing.T) {
+	configMap := map[string]string{
+		"name": "name",
+	}
+	provider := PiccoloTargetProvider{}
+	err := provider.InitWithMap(configMap)
+	assert.Nil(t, err)
+}
+func TestPiccoloTargetProviderInstall(t *testing.T) {
+	testPiccoloProvider := os.Getenv("TEST_PICCOLO_PROVIDER")
+	if testPiccoloProvider == "" {
+		t.Skip("Skipping because TEST_PICCOLO_PROVIDER enviornment variable is not set")
+	}
+	config := PiccoloTargetProviderConfig{}
```

Comment:
There are several places that the service Url needs to be set before the test can pass. We've checked in the mock server to docs/mocks/piccolo-mock folder, please make sure you can test against the server.

Reply:
I haven't paid attention for a while, but thanks for taking the time about this. I've made some changes to Piccolo providers and mock server code, and golang test codes. All tests pass, but since I use podman as my container runtime, others need to fix that part in a docker environment.